### PR TITLE
Change key separator to ||

### PR DESCRIPTION
**Pull request by youngbupark**
# Description

Use a more readable delimiter: ||

## Issue reference

https://github.com/dapr/dapr/issues/914

## Checklist

Please make sure you've  completed the relevant tasks for this PR, out of the following list:

* [x] Code compiles correctly
* [ ] Created/updated tests
* [x] Unit tests passing
* [x] End-to-end tests passing
* [ ] Extended the documentation
* [ ] Specification has been updated
* [ ] Provided sample for the feature

### CODE REVIEW

File: pkg/http/api.go [hunk 1]
```diff
@@ -55,7 +55,7 @@ const (
 	retryPatternParam   = "retryPattern"
 	retryThresholdParam = "retryThreshold"
 	concurrencyParam    = "concurrency"
```
Comment (artursouza) [inline pkg/http/api.go]: I think we decided to keep the old one for http headers.

Comment (youngbupark) [reply pkg/http/api.go]: it is not used by header. https://github.com/dapr/dapr/blob/841d0e6221b6c4f824b392c5531f47df83fe23a0/pkg/http/api.go#L404

Comment (youngbupark) [reply pkg/http/api.go]: @artursouza let me know if I am wrong :)

Comment (artursouza) [reply pkg/http/api.go]: Great!

**Pull request merged**